### DISCUSSION TITLE
Booting hypervisor seems to freeze

### DISCUSSION THREAD

**Discussion by lonnietc** [cloud-hypervisor/cloud-hypervisor#2857] [Q&A]
Hello,

I have an AMD 64 bit system running Ubuntu 20.04 with KVM setup and enabled.

Followed the "2. Getting Started" steps from

https://github.com/cloud-hypervisor/cloud-hypervisor

Everything seems to go very smoothly until I get to the "Run" section and try to run the downloaded focal-server-cloudimg-amd64.raw image:

$ sudo setcap cap_net_admin+ep ./cloud-hypervisor/target/release/cloud-hypervisor
$ ./cloud-hypervisor/target/release/cloud-hypervisor \
	--kernel ./hypervisor-fw \
	--disk path=focal-server-cloudimg-amd64.raw \
	--cpus boot=4 \
	--memory size=1024M \
	--net "tap=,mac=,ip=,mask=" \
	--rng

At this point the system just seems to hang.

![Screenshot from 2021-07-12 07-40-11](https://user-images.githubusercontent.com/15264930/125281665-819e7b00-e2e4-11eb-9996-454e21c3b283.png)

Maybe this is what is supposed to happen, but it does not seem correct to me.

**Comment by sboeuf** [comment]
Hi! It might take some time for your guest to boot, but eventually you should see something like:
```
Ubuntu 20.04.2 LTS cloud hvc0

cloud login:
```

Are you sure you downloaded the latest version of the `hypervisor-fw` binary? Also, you can remove `--rng` since `virtio-rng` is enabled by default.
Last thing, run the same command line with `--serial tty --console off`, this should generate some useful output that might help us understand what's going wrong.

**Reply by yuqitao** [reply]
I got `ubuntu 20.04.4 LTS ubuntu hvc0`，what should I do to login the guest?

**Comment by lonnietc** [comment]
Hello,

I made the changes and got some output from the boot in the serial window:

cloud-hypervisor$ ./cloud-hypervisor/target/release/cloud-hypervisor --kernel hypervisor-fw --disk path=focal-server-cloudimg-amd64.raw --cpus boot=4 --memory size=1024M --net "tap=,mac=,ip=,mask="  --serial tty --console off
Setting up 4 GiB identity mapping
Page tables setup

Booting with PVH Boot Protocol
Found PCI device vendor=8086 device=d57 in slot=0
Found PCI device vendor=1af4 device=1042 in slot=1
Found PCI device vendor=1af4 device=1041 in slot=2
Found PCI device vendor=1af4 device=1044 in slot=3
PCI Device: 0:1.0 1af4:1042
Bar: type=MemorySpace32 address=e7f80000
Bar: type=MemorySpace32 address=0
Bar: type=MemorySpace32 address=0
Bar: type=MemorySpace32 address=0
Bar: type=MemorySpace32 address=0
Bar: type=Unused address=0
Virtio block device configured. Capacity: 4612096 sectors
Found EFI partition
Filesystem ready
Error loading default entry: FileError(NotFound)
Using EFI boot.
Found bootloader (BOOTX64.EFI)
Executable loaded
TPM logging failed: Unsupported
System BootOrder not found.  Initializing defaults.
Creating boot entry "Boot0000" with label "ubuntu" for file "\EFI\UBUNTU\shimx64.efi"

TPM logging failed: Unsupported

any ideas as to what this might be?
cheers
Lonnie

**Reply by sboeuf** [reply]
You might be using a focal image that doesn't contain the right things. Please try with the one from the CI scripts:
```
wget https://cloud-hypervisor.azureedge.net/focal-server-cloudimg-amd64-custom-20210609-0.qcow2
```
then you can convert it to `RAW` if needed with the following command:
```
qemu-img convert -p -f qcow2 -O raw focal-server-cloudimg-amd64-custom-20210609-0.qcow2 focal-server-cloudimg-amd64-custom-20210609-0.raw
```

**Comment by lonnietc** [comment]
I'm working on the download now but the connection seems to keep dropping and reconnecting.

Also, I did find some other x86_64 qcow2 images:

Centos:  https://cloud.centos.org/centos/8/x86_64/images/

So, in theory, I should be able to run other smaller x86_64 Linux images as well, right?
Cheers

**Comment by lonnietc** [comment]
Still not able to get the image from your link as the connections still keeps dropping off.

I was wondering if it might be possible to boot ISO images with cloud-hypervisor?

I have many of them for 64-bit and 32-bit (although Cloud-Hypervisor does not support 32-bit platforms).
cheers

**Comment by lonnietc** [comment]
Hello,

I was going to try to run one of these images:

As of 2021-04-29, the following cloud images are supported:

1. Ubuntu Bionic (cloudimg)  https://cloud-images.ubuntu.com/bionic/current/

2. Ubuntu Focal (cloudimg)  https://cloud-images.ubuntu.com/focal/current/

3. Ubuntu Groovy (cloudimg)  https://cloud-images.ubuntu.com/groovy/current/

4. Ubuntu Hirsute (cloudimg)  https://cloud-images.ubuntu.com/hirsute/current/

I grabbed one from:

https://cloud-images.ubuntu.com/bionic/current/bionic-server-cloudimg-amd64.img 

Then converted it with qemu-image:

qemu-img convert -p -f qcow2 -O raw bionic-server-cloudimg-amd64.img bionic-server-cloudimg-amd64.raw

then tried to run it with cloud-hypervisor:

:~/cloud-hypervisor$ ./cloud-hypervisor/target/x86_64-unknown-linux-musl/release/cloud-hypervisor --kernel hypervisor-fw --disk path=bionic-server-cloudimg-amd64.raw --cpus boot=4 --memory size=2048M --net "tap=,mac=,ip=,mask="  --serial tty --console off

but that did not work either:

![Screenshot from 2021-07-12 14-57-10](https://user-images.githubusercontent.com/15264930/125341397-8e8d8f80-e321-11eb-82f6-3f04c20f3b91.png)

Any ideas?

Probably I am missing something simple here, but do not yet seem to have the hang of it.

Aside:   Will Cloud-Hypervisor be able to boot ISO's anytime soon?

Cheers,
Lonnie

**Reply by sboeuf** [reply]
I've just tried your setup and it works fine for me. The point is, you have to wait for a minute to get it booting correctly. You get these "errors" at first but then it boots up to this point:
```
[  OK  ] Reached target Login Prompts.
[  OK  ] Started LSB: automatic crash report generation.
[  OK  ] Started LSB: Record successful boot for GRUB.
[  OK  ] Started LXD - container startup/shutdown.
[  OK  ] Started Pollinate to seed the pseudo random number generator.
         Starting OpenBSD Secure Shell server...
[  OK  ] Started Dispatcher daemon for systemd-networkd.
[  OK  ] Started OpenBSD Secure Shell server.
[  OK  ] Started Snap Daemon.
         Starting Wait until snapd is fully seeded...

Ubuntu 18.04.5 LTS ubuntu ttyS0

ubuntu login:
```

About ISO, we don't have plan to support it anytime soon, but you're more than welcome to submit some patches to add the support if you need it :)

**Reply by lonnietc** [reply]
Perhaps it is something wrong with my AMD system and KVM as I let it sit for 30-minutes and it still did not boot up.

**Reply by sboeuf** [reply]
Try with another image maybe.

**Reply by rbradford** [reply]
I don't believe anyone has validated Cloud Hypervisor on KVM on AMD recently and we have no automated coverage of it. If you have the skill set you could go back to a very early Cloud Hypervisor release and attempt to git bisect from there if it succeeds. Be aware that older versions of Cloud Hypervisor have different command line options and I would recommend you only test with direct kernel boot.

**Reply by iggy** [reply]
I use cloud-hypervisor on an AMD system (Ryzen 5900) with direct kernel boot (using virtiofs as root). It works fine. I'll try to test using a disk image later, but it definitely was working within the last couple versions.

**Reply by rbradford** [reply]
@iggy That's really good to know. Thank you! (I was concerned as I know some of the CPUID leaves are different for AMD vs Intel and we've made changes in that relatively recently.)

**Comment by iggy** [comment]
I just tested the latest focal kvm image with cloud-hypervisor v16 with rust-hypervisor-firmware 0.3.1 and it boots up fine on my Ryzen 5900.

Might be interesting to know what kind of system is actually being used. I feel like I remember issues with qemu/kvm and older AMD cpus (bulldozer era) being reportedly recently.

If you ever need anything else AMD tested, feel free to reach out on here/slack/etc.
